src/awesomeSezzle.js

```diff
@@ -675,25 +675,30 @@ class AwesomeSezzle {
 		return currency || 36;
 	}
 
-	calculateInterest(price, term, APR) {
+	calculateMonthlyWithInterest(price, term, APR) {
 		if(APR > 0){
 			const rate = (APR/100)/12;
 			const numerator = price * rate * Math.pow(1+rate, term);
 			const denominator = Math.pow(1+rate, term)-1;
 			const interestPayment = numerator/denominator
 			return interestPayment;
 		} else {
-			return price/term;
+			return (price/term).toFixed(2);
 		}
 	}
-	calculateMonthly(priceString, parseMode, term, APR) {
-		const interestAmount = this.calculateInterest(priceString, term, APR);
-		return this.addDelimiters(interestAmount.toFixed(2), parseMode);
+	formatMonthly(priceString, parseMode, term, APR) {
+		const interestAmount = this.calculateMonthlyWithInterest(priceString, term, APR);
+		return this.addDelimiters(interestAmount, parseMode);
 	}
-	calculateAdjusted(priceString, parseMode, term, APR) {
-		const interestAmount = this.calculateInterest(priceString, term, APR);
-		return this.addDelimiters((interestAmount*term).toFixed(2), parseMode);
+	formatTotalInterest(priceString, parseMode, term, APR) {
+		const adjustedTotal = this.calculateMonthlyWithInterest(priceString, term, APR) * term;
+		return this.addDelimiters(adjustedTotal-priceString, parseMode);
 	}
+	formatAdjustedTotal(priceString, parseMode, term, APR) {
+		const interestAmount = this.calculateMonthlyWithInterest(priceString, term, APR);
+		return this.addDelimiters((interestAmount*term), parseMode);
```

Comment:
The logic of calculating the total (which is in `formatTotalInterest`) is being repeated here

Reply:
The addDelimiters function stringifies the amount and changes the delimiters for EU format.

Reply:
I am talking about the `interestAmount*term` part here. It is basically the same logic as on the line 694, but rewritten and embedded into the function input parameter.